shared/modules/doc/README.md

```diff
@@ -1,53 +1,67 @@
-# Genesis Module
+# Modules <!-- omit in toc -->
 
-This document is meant to be the development level documentation for the genesis state details related to the design of the codebase and information related to development.
+This document outlines how we structured the code by splitting it into modules, what a module is and how to create one.
 
-!!! IMPORTANT !!!
+## Contents <!-- omit in toc -->
 
-This directory was created for the purposes of integration between the four core modules and is
-not intended to store all the core shared types in the long-term.
+- [Definitions](#definitions)
+	- [**Module common interface**](#module-common-interface)
+	- [**Module**](#module)
+	- [**Module mock**](#module-mock)
+	- [**Base module**](#base-module)
+- [Code Organization](#code-organization)
+- [Modules in detail](#modules-in-detail)
+	- [Module creation](#module-creation)
+	- [Interacting \& Registering with the `bus`](#interacting--registering-with-the-bus)
+		- [Modules Registry](#modules-registry)
+	- [Start the module](#start-the-module)
+	- [Add a logger to the module](#add-a-logger-to-the-module)
+	- [Get the module `bus`](#get-the-module-bus)
+	- [Stop the module](#stop-the-module)
 
-Speak to @andrewnguyen22 or @Olshansk for more details.
+## Definitions
 
-!!! IMPORTANT !!!
+### **Module common interface**
 
-## Implementation
+A module common interface is an interface that defines the methods that we modelled as common to multiple modules. For example: the ability to start/stop a module is pretty common and for that we defined the `InterruptableModule` interface.
```

Comment:
NIT: "shared module interfaces" sounds more intuitive, but maybe you were intending "module common interface" for a reason?